# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  70268b8713c13297cf869be55951c7614dad3c1a
Misc: Show identity and server keys in the troubleshooting information.

diff --git a/mail/components/about-support/content/accounts.js b/mail/components/about-support/content/accounts.js
--- a/mail/components/about-support/content/accounts.js
+++ b/mail/components/about-support/content/accounts.js
@@ -63,18 +63,18 @@ var AboutSupport = {
   _getSMTPDetails(aAccount) {
     let defaultIdentity = aAccount.defaultIdentity;
     let smtpDetails = [];
 
     for (let identity of aAccount.identities) {
       let isDefault = identity == defaultIdentity;
       let smtpServer = MailServices.smtp.getServerByIdentity(identity);
       smtpDetails.push({
-        identityName: identity.identityName,
-        name: smtpServer.displayname,
+        identityName: `${identity.identityName} (${identity.key})`,
+        name: `${smtpServer.displayname} (${smtpServer.key})`,
         authMethod: smtpServer.authMethod,
         socketType: smtpServer.socketType,
         isDefault,
       });
     }
 
     return smtpDetails;
   },
@@ -86,21 +86,17 @@ var AboutSupport = {
     let accountDetails = [];
 
     for (let account of MailServices.accounts.accounts) {
       let server = account.incomingServer;
       accountDetails.push({
         key: account.key,
         name: server.prettyName,
         hostDetails:
-          "(" +
-          server.type +
-          ") " +
-          server.hostName +
-          (server.port != -1 ? ":" + server.port : ""),
+          `(${server.type}) ${server.hostName}${(server.port != -1 ? ":" + server.port : "")} (${server.key})`,
         socketType: server.socketType,
         authMethod: server.authMethod,
         smtpServers: this._getSMTPDetails(account),
       });
     }
 
     function idCompare(accountA, accountB) {
       let regex = /^account([0-9]+)$/;
